Gems/ROS2/Code/Source/RobotImporter/URDF/SensorsMaker.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "SensorsMaker.h"
+
+#include <AzCore/Component/EntityId.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+
+#include <ROS2/RobotImporter/SDFormatSensorImporterHook.h>
+
+#include <sdf/Link.hh>
+#include <sdf/Sensor.hh>
+
+namespace ROS2
+{
+    void AddSensor(AZ::EntityId entityId, const sdf::Sensor* sensor)
+    {
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        auto serializeContext = AZ::Interface<AZ::ComponentApplicationRequests>::Get()->GetSerializeContext();
+        serializeContext->EnumerateAll(
+            [&](const AZ::SerializeContext::ClassData* classData, const AZ::Uuid& typeId) -> bool
+            {
+                auto* attribute = AZ::FindAttribute(AZ::Crc32("SensorImporterHooks"), classData->m_attributes);
+                if (attribute == nullptr)
+                {
+                    return true;
+                }
+
+                AZ::AttributeReader reader(nullptr, attribute);
+                AZStd::vector<ROS2::SDFormat::SensorImporterHook> sensorHooks;
+                if (reader.Read<AZStd::vector<ROS2::SDFormat::SensorImporterHook>>(sensorHooks))
+                {
+                    for (const auto& hook : sensorHooks)
+                    {
+                        for (const auto& t : hook.m_sensorTypes)
```

Comment:
m_sensorTypes is ` AZStd::unordered_set<sdf::SensorType> `. Please use find/contain instead of loop.